src/app/shared/services/auth/providers/firebase.auth.ts

```diff
@@ -67,10 +78,20 @@ export class FirebaseAuthProvider extends AuthProviderBase {
     const authUser: IAuthUser = {
       ...profile,
       uid: user.uid,
+      name: user.displayName,
     };
     this.authUser.set(authUser);
   }
 
+  private saveUserInfo(user: User, additionalUserInfo: AdditionalUserInfo) {
+    // NOTE - additionalUserInfo is only returned on first signIn so persist to localStorage
+    // for access on automated sign-in following restart. Use fallback empty object if null
+    const { profile = {} } = additionalUserInfo;
+    localStorage.setItem(AUTH_METADATA_FIELD, JSON.stringify(profile));
+
+    this.setAuthUser(user, profile);
```

Comment:
nit(blocking): I can see how `setAuthUser` is required by both however I think better to duplicate the code in the signIn methods instead of moving it into the common function for better separation of concerns (the method would otherwise need to be called `saveUserInfoAndSetAuthUser`

Also would be good to preserve the operation order in case of knock-ons (previously was set then save, here it is save then set, which could throw error before user set if issues saving profile)

Reply:
Yeh that makes sense about duplicating the code, my instincts said the same thing.

I believe I did originally preserve the order (save then set), but for the reason you've suggested I think it makes sense to do set then save.

See updates in https://github.com/IDEMSInternational/open-app-builder/pull/2869/commits/8bdb4c98f6386fb4e0e669a2eab48cbe66b9d3d1